pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1704,12 +1705,18 @@ func formatDomainCPUTune(vmi *v1.VirtualMachineInstance, domain *api.Domain, c *
 	if len(c.CPUSet) == 0 {
 		return fmt.Errorf("failed for get pods pinned cpus")
 	}
+	CPUSet, err := hardware.GroupCPUThreads(c.CPUSet)
+	if err != nil {
+		log.Log.Reason(err).Warning("Failed to get cpu topology")
+		CPUSet = c.CPUSet
+	}
```

Comment:
Good point. Initially I was thinking about it but then decided to make the change less "intrusive" and to leave the original `c.CPUSet` unmodified. But I think you are right and it can be moved to `manager.go`.